src/main.rs

```diff
@@ -76,18 +76,18 @@ fn main() -> Result<()> {
         if let Some(text) = text {
             input = format!("{text}\n{input}");
         }
-        start_directive(client, config, &input, no_stream)
+        start_directive(&client, &config, &input, no_stream)
     } else {
         match text {
-            Some(text) => start_directive(client, config, &text, no_stream),
+            Some(text) => start_directive(&client, &config, &text, no_stream),
             None => start_interactive(client, config),
         }
     }
 }
 
 fn start_directive(
-    client: ChatGptClient,
-    config: SharedConfig,
+    client: &ChatGptClient,
+    config: &SharedConfig,
     input: &str,
     no_stream: bool,
 ) -> Result<()> {
```

Comment:
(this happens in quite a few places, so I'll just acknowledge it here)

These parameters were passed by value, but never consumed. It is often more flexible to pass by reference, and this can also prevent unnecessary allocations.

https://rust-lang.github.io/rust-clippy/master/index.html#/needless_pass_by_value